### PR TITLE
feat(plug): adds plug for validation on routes

### PR DESCRIPTION
**Pull request by mgartner**
#### What

* Adds a Plug for validating requests
* Please read the README for details on how it can be used

#### Why

* I was integrating Litmus into one of our services, and it occured to me that having middleware to do validation would drastically simplify things.

#### Questions

* Does this interface make sense to you?
* Anyway you can think to make it easier to use?
* Suggestions on the naming of options?

### CODE REVIEW

File: test/litmus/plug_test.exs [hunk 1]
```diff
@@ -0,0 +1,86 @@
+defmodule Litmus.PlugTest do
+  use ExUnit.Case, async: true
+  use Plug.Test
+
+  describe "init/1" do
+    test "returns default histogram function when passed an empty array" do
```
Comment (kylekwong) [inline test/litmus/plug_test.exs]: What's this default histogram function?

**Review by kylekwong — APPROVED** [review]
just one question

File: test/litmus/plug_test.exs (hunk 1 above)
Comment (mgartner) [reply test/litmus/plug_test.exs]: sorry, copy pasta 🍝

**Review by jhwang09 — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -184,6 +184,50 @@ iex> Litmus.validate(params, schema)
...
+    "id" => %Litmus.Type.Number{
+      required: true
+    }
+  }
+
+  get "/test", private: %{litmus_query: @schema} do
+    Plug.Conn.send_resp(conn, 200, "items")
+  end
+
+  post "/test", private: %{litmus_body: @schema} do
```
Comment (jhwang09) [inline README.md:219]: didn't know you can do this, good to learn!

Comment (mgartner) [reply README.md:219]: i didn't know either until i started working on this lol

**Pull request merged**